### PR TITLE
Feat:Compatible with Dify's External Knowledge API

### LINKED ISSUE #2731: [Feature Request]: Please Build a Retrieval API
### Is there an existing issue for the same feature request?

- [X] I have checked the existing issues.

### Is your feature request related to a problem?

_No response_

### Describe the feature you'd like

I want to integrate Ragflow's knowledge base and Dify. 

Ragflow specializes in building robust RAG, while Dify specializes in building powerful agent.

I want to leverage the strengths of both platforms via API.

But I can't find a retrieval API in Ragflow's docs.

So I want to kindly ask Ragflow's team to build a retrieval API.

This will make the AI agent revolution take a huge step forward!

https://docs.dify.ai/guides/knowledge-base/external-knowledge-api-documentation

### Describe implementation you've considered

_No response_

### Documentation, adoption, use case

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Howe829**
### What problem does this PR solve?

_Briefly describe what this PR aims to solve. Include background context that will help reviewers understand the purpose of the PR._
Fixes #2731 
### Type of change

- [x] New Feature (non-breaking change which adds functionality)

### CODE REVIEW

File: api/apps/sdk/retrieval.py
Comment (KevinHuSh) [inline api/apps/sdk/retrieval.py:1]: Could you change file name to dify_retrieval,?

File: api/apps/sdk/retrieval.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from flask import request, jsonify
+
+from db import LLMType, ParserType
+from db.services.knowledgebase_service import KnowledgebaseService
+from db.services.llm_service import LLMBundle
+from settings import retrievaler, kg_retrievaler, RetCode
+from utils.api_utils import validate_request, build_error_result, apikey_required
+
+
+@manager.route('/retrieval', methods=['POST'])
```
Comment (KevinHuSh) [inline api/apps/sdk/retrieval.py]:
```suggestion
@manager.route('/dify_retrieval', methods=['POST'])
```

Comment (Howe829) [reply api/apps/sdk/retrieval.py]: image
Thanks for your comment but the api request url needs to be "endpoint/retrieval".
https://docs.dify.ai/guides/knowledge-base/external-knowledge-api-documentation

Comment (Howe829) [reply api/apps/sdk/retrieval.py]: I've added dify to url prefix and changed the filename, PTAL.

**Comment by nicklhy** [comment]
@Howe829 Hi, I just did a few tests of this retrieval api in dify and opened an issue [here](https://github.com/langgenius/dify/issues/9448). It seems the type of `metadata` was wrongly written in dify's doc as string. Indeed, it should be a dict. 

Could you please repair the code here ? I guess simply set `metadict = {}` is ok ?

https://github.com/langgenius/dify/issues/9448#issuecomment-2421234882

**Comment by Howe829** [comment]
> @Howe829 Hi, I just did a few tests of this retrieval api in dify and opened an issue [here](https://github.com/langgenius/dify/issues/9448). It seems the type of `metadata` was wrongly written in dify's doc as string. Indeed, it should be a dict.
> 
> Could you please repair the code here ? I guess simply set `metadict = {}` is ok ?
> 
> [langgenius/dify#9448 (comment)](https://github.com/langgenius/dify/issues/9448#issuecomment-2421234882)

Sorry for the bug.I am quit busy recently, so I have no time to deal with it. Thanks for your fix!

**Comment by nicklhy** [comment]
already solved in https://github.com/infiniflow/ragflow/pull/3076

**Pull request merged**